pkg/detectors/jdbc/postgres.go

```diff
@@ -95,46 +102,51 @@ func parsePostgres(ctx logContext.Context, subname string) (jdbc, error) {
 		// https://www.postgresql.org/docs/current/libpq-ssl.html#LIBPQ-SSL-PROTECTION
 		case "disable", "allow", "prefer",
 			"require", "verify-ca", "verify-full":
-			params["sslmode"] = v[0]
+			postgresJDBC.Params["sslmode"] = v[0]
 		}
 	}
 
 	if v := u.Query().Get("user"); v != "" {
-		params["user"] = v
+		postgresJDBC.User = v
 	}
 
 	if v := u.Query().Get("password"); v != "" {
-		params["password"] = v
+		postgresJDBC.Password = v
 	}
 
-	if params["host"] == "" || params["password"] == "" {
+	if postgresJDBC.Host == "" || postgresJDBC.Password == "" {
 		ctx.Logger().WithName("jdbc").
 			V(2).
 			Info("Skipping invalid Postgres URL - no password or host found")
 		return nil, fmt.Errorf("missing host or password in connection string")
 	}
 
-	return &postgresJDBC{subname[2:], params}, nil
+	return postgresJDBC, nil
 }
 
-func buildPostgresConnectionString(params map[string]string, includeDbName bool) string {
+func BuildPostgresConnectionString(host string, user string, password string, dbName string, params map[string]string, includeDbName bool) string {
 	data := map[string]string{
 		// default user
-		"user": "postgres",
+		"user":     "postgres",
+		"password": password,
+		"host":     host,
+	}
+	if user != "" {
+		data["user"] = user
+	}
+	if h, p, found := strings.Cut(host, ":"); found {
+		data["host"] = h
+		data["port"] = p
 	}
```

Comment:
As per the [coding guidelines](https://docs.google.com/document/d/1clyphtrspL_e0IhVPONtBpl65bjGMYLiBjkEu5UYmrE/edit?tab=t.0)
```suggestion
	if h, p, ok := strings.Cut(host, ":"); ok {
		data["host"] = h
		data["port"] = p
}
```